packages/forms/test/reactive_integration_spec.ts

```diff
@@ -2733,6 +2733,80 @@ const ValueAccessorB = createControlValueAccessor('[cva-b]');
             expect(form.controls.pin.errors).toEqual({max: {max: -10, actual: 0}});
           });
         });
+
+        it('should fire registerOnValidatorChange for validators attached to the formGroups', () => {
+          let registerOnValidatorChangeFired = 0;
+
+          @Directive({
+            selector: '[ng-noop-validator]',
+            providers: [{
+              provide: NG_VALIDATORS,
+              useExisting: forwardRef(() => NoOpValidator),
+              multi: true
+            }]
+          })
+          class NoOpValidator implements Validator {
+            @Input() injectedInput = '';
+
+            validate(c: AbstractControl) {
+              return null;
+            }
+
+            public registerOnValidatorChange(fn: () => void) {
+              registerOnValidatorChangeFired++;
+            }
+          }
+
+          let registerOnAsyncValidatorChangeFired = 0;
+
+          @Directive({
+            selector: '[ng-noop-async-validator]',
+            providers: [{
+              provide: NG_ASYNC_VALIDATORS,
+              useExisting: forwardRef(() => NoOpAsyncValidator),
+              multi: true
+            }]
+          })
+          class NoOpAsyncValidator implements AsyncValidator {
+            @Input() injectedInput = '';
+
+            validate(c: AbstractControl) {
+              return Promise.resolve(null);
+            }
+
+            public registerOnValidatorChange(fn: () => void) {
+              registerOnAsyncValidatorChangeFired++;
+            }
+          }
+
+          @Component({
+            selector: 'ng-model-noop-validation',
+            template: `
+            <form [formGroup]="fooGroup" ng-noop-validator ng-noop-async-validator [injectedInput]="ei.value">
+                <input type="text" formControlName="fooInput" #ei>
+            </form>
+           `
+          })
+          class NgModelNoOpValidation {
+            fooGroup = new FormGroup({
+              fooInput: new FormControl(''),
+            });
+          }
```

Comment:
I'd propose simplifying this a bit and avoid using local references (i.e. `#ei`) and just use a class field as a bound property:
```suggestion
          @Component({
            selector: 'ng-model-noop-validation',
            template: `
            <form [formGroup]="fooGroup" ng-noop-validator ng-noop-async-validator [validatorInput]="validatorInput">
                <input type="text" formControlName="fooInput">
            </form>
           `
          })
          class NgModelNoOpValidation {
            validatorInput: string = 'foo';
          
            fooGroup = new FormGroup({
              fooInput: new FormControl(''),
            });
          }
```
and then you can trigger the input change by doing the following:

```
  fixture.componentInstance.validatorInput = 'bar';
  fixture.detectChanges();
```

Reply:
Done, in both tests.